src/core/server/plugins/plugins_service.ts

```diff
@@ -196,6 +207,10 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
             throw new Error(`Plugin with id "${plugin.name}" is already registered!`);
           }
 
+          if (plugin.includesUiPlugin) {
+            this.uiPluginBundlePaths.set(plugin.name, `${plugin.path}/public`);
+          }
```

Comment:
Even if the path was the only thing actually used in our `internal` plugin map, are we really sure we want to convert this to a map of endpoint path strings? This seems like less flexible and futur-proof in case we need to add additional internal metadatas for plugins.

Reply:
Maybe it would make more sense to combine all of these values into a single Map structure with 3 properties: publicManifest, browserConfig$, and entryPointPath.

Reply:
Though the manifest and browserConfig keys would be optional since they'd only be present for enabled plugins. That may actually make consuming these a bit more awkward?

Reply:
I think keeping them separate is a good thing, as they are used in different use cases. 

Was just wondering/asking if removing the `internal` map for this new `uiPluginBundlePaths` was really justified (could have just kept the `internalPlugin` structure with only the `bundlePath` properties, would have allowed to add additional internal properties if we need to in the futur), but that works for now, we can change it later if the need arises.

Reply:
I'm happy to put it back under internal. Seems safer to make that clear that this shouldn't be exposed to the client.